src/cromshell/execution_status_count/command.py

```diff
@@ -0,0 +1,251 @@
+
+    # Determine what color to print task print out
+    if shards_failed == 0 and shards_running == 0:
+        task_status_font = io_utils.TextStatusesColor.TASK_COLOR_SUCCEEDED
+    elif shards_failed > 0 and shards_running > 0:
+        task_status_font = io_utils.TextStatusesColor.TASK_COLOR_FAILING  # Running but will fail
+    elif shards_running > 0:
+        task_status_font = io_utils.TextStatusesColor.TASK_COLOR_RUNNING
+    else:
+        task_status_font = io_utils.TextStatusesColor.TASK_COLOR_FAILED
+
+    print(
+        colored(
+            f"{indent}{task}\t{shards_running} Running, {shards_done} Done, {shards_retried} Preempted, {shards_failed} Failed",
+            color=task_status_font,
+        )
+    )
+
+    # If the task has shards that failed list them
+    # Maybe place contents of if statement in function and add indentation to printout??
+    if shards_failed:
+        print_list_of_failed_shards(
+            shards=shards, indent=indent, task_status_font=task_status_font
+        )
+
+
+def print_task_status_summary(workflow_metadata: dict) -> None:
+    """
+    Prints the status count for each task in a workflow.
+    Does NOT run expose subworkflows
+
+    :param workflow_metadata: Metadata of the workflow to process
+    :return:
+    """
+
+    workflow_status_summary = {}
+    for task in workflow_metadata["calls"]:
+        # Method to sort task shards by status (executionStatus)
+        shards = workflow_metadata["calls"][task]
+
+        # Add the status counts for this task to dictionary holding other
+        # task status counts
+        workflow_status_summary[task] = get_shard_status_count(shards)
+
+    io_utils.pretty_print_json(format_json=workflow_status_summary)
+
+
+def get_shard_status_count(shards: dict) -> dict:
+    """
+    Count the number of shards for each status type and return as dictionary.
+    :param shards: Task shards
+    :return:
+    """
+
+    sorted_shards = sorted(shards, key=lambda y: y['executionStatus'])
+    statuses_count = {}
+    for status, group in groupby(sorted_shards, lambda x: x["executionStatus"]):
+        statuses_count[status] = len(list(group))
+
+    return statuses_count
+
+
+def print_list_of_failed_shards(
+        shards: dict, indent: str, task_status_font: str
+) -> None:
+    """
+    Print a list of the failed shards
+    :param shards: The shared of a called task
+    :param indent: Indent string given as "\t", used to indent print out
+    :param task_status_font: Color of the print out
+    :return:
+    """
+    failed_shards = []
+    failed_shards_index = []
+    sorted_shards = sorted(shards, key=lambda y: y["executionStatus"])
+    for status, group in groupby(sorted_shards, lambda x: x["executionStatus"]):
+        if status == "Failed":
+            failed_shards = list(group)
+
+    for shard in failed_shards:
+        failed_shards_index.append(shard["shardIndex"])
+    print(
```

Comment:
Along with printing, might be good to have the function return the printout to make it easier when testing

Reply:
done